app/src/main/java/org/mozilla/focus/search/ManualAddSearchEnginePreference.java

```diff
@@ -49,9 +54,37 @@ protected View onCreateView(ViewGroup parent) {
         searchQueryEditText.addTextChangedListener(buildTextWatcherForErrorLayout(searchQueryErrorLayout));
 
         progressView = view.findViewById(R.id.progress);
+
+        updateState();
         return view;
     }
 
+    @Override
+    protected void onRestoreInstanceState(Parcelable state) {
+        SavedState savedState = (SavedState) state;
+        super.onRestoreInstanceState(savedState.getSuperState());
+        savedSearchEngineName = savedState.searchEngineName;
+        savedSearchQuery = savedState.searchQuery;
+    }
+
+    @Override
+    protected Parcelable onSaveInstanceState() {
+        Parcelable superState = super.onSaveInstanceState();
+        SavedState savedState = new SavedState(superState);
+        savedState.searchEngineName = engineNameEditText.getText().toString();
+        savedState.searchQuery = searchQueryEditText.getText().toString();
+        return savedState;
+    }
+
+    private void updateState() {
+        if (savedSearchEngineName != null) {
```

Comment:
I think it would be better to null-check on the `engineNameEditText` and the `searchQueryEditText` themselves, as if you do not and the `EditText` is null, you'll crash the app; while if you simply set a `null` `String` on an existing `EditText` you'll see an empty text in the worst case scenario

Reply:
Thanks, fixed it :)